File: Android App/app/src/main/java/revert/shredsheets/models/scales/HarmonicMinor.java
package revert.shredsheets.models.scales;

public class HarmonicMinor extends Scale {

    @Override
    public int[] getScaleIntervals() {
        return new int[]{2, 1, 2, 2, 1, 3, 1};
    }

    @Override
    public String getName() {
        return "Harmonic Minor";
    }

    @Override
    public String[] getModeNames() {
        return new String[]{"Harmonic Minor", "Locrian ♯6", "Ionian Augmented", "Romanian", "Phrygian Dominant", "Lydian ♯2", "UltraLocrian"};
    }

    @Override
    public int[][] getDegrees() {
        return new int[][]{
                {1, 2, 3, 4, 5, 6, 7},
                {1, 2, 3, 4, 5, 6, 7},
                {1, 2, 3, 4, 5, 6, 7},
                {1, 2, 3, 4, 5, 6, 7},
                {1, 2, 3, 4, 5, 6, 7},
                {1, 2, 3, 4, 5, 6, 7},
                {1, 2, 3, 4, 5, 6, 7}
        };
    }

    @Override
    public String[] getChords() {
        return new String[]{"mΔ7", "m7 ♭5", "M7+", "m7", "7", "M7", "m7o"};
    }
}

File: Shredsheets 3/core/src/com/bm/shredsheets/models/scales/HarmonicMinor.kt
package com.bm.shredsheets.models.scales

class HarmonicMinor : Scale() {
    override val scaleIntervals: IntArray
        get() = intArrayOf(2, 1, 2, 2, 1, 3, 1)

    override val name: String
        get() = "Harmonic Minor"

    override fun getModeNames(): Array<String> {
        return arrayOf("Harmonic Minor", "Locrian ♯6", "Ionian Augmented", "Romanian", "Phrygian Dominant", "Lydian ♯2", "UltraLocrian")
    }

    override val getEnum: Scales
        get() = Scales.HarmonicMinor

    override val degrees: Array<IntArray>
        get() = arrayOf(
            intArrayOf(1, 2, 3, 4, 5, 6, 7),
            intArrayOf(1, 2, 3, 4, 5, 6, 7),
            intArrayOf(1, 2, 3, 4, 5, 6, 7),
            intArrayOf(1, 2, 3, 4, 5, 6, 7),
            intArrayOf(1, 2, 3, 4, 5, 6, 7),
            intArrayOf(1, 2, 3, 4, 5, 6, 7),
            intArrayOf(1, 2, 3, 4, 5, 6, 7))

    override val chords: Array<String>
        get() = arrayOf("mΔ7", "m7 ♭5", "M7+", "m7", "7", "M7", "m7o")
}